src/utils/access_log.ts

```diff
@@ -128,17 +130,24 @@ export const createAwsS3AccessLogWriter = (
   blobName: string
 ): ReturnType<AccessLogWriter> =>
   pipe(
-    TE.tryCatch(
-      async () =>
-        await s3Service.send(
-          new PutObjectCommand({
-            Body: JSON.stringify(encryptedBlobItem),
-            Bucket: bucketName,
-            Key: blobName
-          })
-        ),
-      E.toError
-    ),
+    TE.tryCatch(async () => {
+      const body = JSON.stringify(encryptedBlobItem);
+      const md5 = md.md5.create();
+      md5.update(body);
+      const md5Digest = md5.digest().toHex();
+      const contentMD5: string = Buffer
+        .from(md5Digest, "hex")
+        .toString("base64");
```

Comment:
I think that extract this code lines in a separate function make the code more readable.